agent/agent.go

```diff
@@ -334,30 +330,6 @@ func LocalConfig(cfg *config.RuntimeConfig) local.Config {
 	return lc
 }
 
-func (a *Agent) setupProxyManager() error {
-	acfg, err := a.config.APIConfig(true)
-	if err != nil {
-		return fmt.Errorf("[INFO] agent: Connect managed proxies are disabled due to providing an invalid HTTP configuration")
-	}
-	a.proxyManager = proxyprocess.NewManager()
-	a.proxyManager.AllowRoot = a.config.ConnectProxyAllowManagedRoot
-	a.proxyManager.State = a.State
-	a.proxyManager.Logger = a.logger
-	if a.config.DataDir != "" {
-		// DataDir is required for all non-dev mode agents, but we want
-		// to allow setting the data dir for demos and so on for the agent,
-		// so do the check above instead.
-		a.proxyManager.DataDir = filepath.Join(a.config.DataDir, "proxy")
```

Comment:
Should we retain some transitional code that looks for the `proxy` subdirectory and cleans it up now that the feature is going to be killed off?

Reply:
Destructive updates like proactively deleting configuration feels like something I'd rather avoid? I can see how it could be confusing if debugging to find config that's no longer in use, but I'm thinking about like if there's a failed upgrade attempt and a user needs to roll back, I wouldn't want to impact their ability to do so?

Reply:
I'm OK with not cleaning up. It's not going to be any real disk space and it was deprecated as a beta feature.

Reply:
Maybe documenting how to manually clean up this directory once it's no longer in use would be sufficient?